src/Apps/Order/Routes/NewPayment/index.tsx

```diff
@@ -0,0 +1,496 @@
+import * as Schema from "Artsy/Analytics/Schema"
+import { CountdownTimer } from "Components/v2/CountdownTimer"
+import { Router } from "found"
+import React, { Component } from "react"
+import {
+  commitMutation,
+  createFragmentContainer,
+  graphql,
+  RelayRefetchProp,
+} from "react-relay"
+import { injectStripe, ReactStripeElements } from "react-stripe-elements"
+import { ErrorWithMetadata } from "Utils/errors"
+import createLogger from "Utils/logger"
+import { Media } from "Utils/Responsive"
+
+import {
+  Button,
+  Checkbox,
+  Col,
+  Collapse,
+  Flex,
+  Join,
+  Row,
+  Serif,
+  Spacer,
+} from "@artsy/palette"
+import { Dialog, injectDialog } from "Apps/Order/Dialogs"
+
+export const ContinueButton = props => (
+  <Button size="large" width="100%" {...props}>
+    Continue
+  </Button>
+)
+
+export interface NewPaymentProps
+  extends ReactStripeElements.InjectedStripeProps {
+  order: NewPayment_order
+  relay?: RelayRefetchProp
+  router: Router
+  dialog: Dialog
+}
+
+interface NewPaymentState {
+  hideBillingAddress: boolean
+  address: Address
+  addressErrors: AddressErrors
+  addressTouched: AddressTouched
+  stripeError: stripe.Error
+  isCommittingMutation: boolean
+}
+
+const logger = createLogger("Order/Routes/NewPayment/index.tsx")
+
+@track()
+export class NewPaymentRoute extends Component<
+  NewPaymentProps,
+  NewPaymentState
+> {
+  state = {
+    hideBillingAddress: true,
+    stripeError: null,
+    isCommittingMutation: false,
+    address: this.startingAddress(),
+    addressErrors: {},
+    addressTouched: {},
+  }
+
+  startingAddress(): Address {
+    return {
+      ...emptyAddress,
+      country: "US",
+    }
+  }
+
+  get touchedAddress() {
+    return {
+      name: true,
+      country: true,
+      postalCode: true,
+      addressLine1: true,
+      addressLine2: true,
+      city: true,
+      region: true,
+      phoneNumber: true,
+    }
+  }
+
+  onContinue: () => void = () => {
+    this.setState({ isCommittingMutation: true }, () => {
+      if (this.needsAddress()) {
+        const { errors, hasErrors } = validateAddress(this.state.address)
+        if (hasErrors) {
+          this.setState({
+            isCommittingMutation: false,
+            addressErrors: errors,
+            addressTouched: this.touchedAddress,
+          })
+          return
+        }
+      }
+
+      const { address } = this.state
+      const stripeBillingAddress = this.getStripeBillingAddress(address)
+      this.props.stripe
+        .createToken(stripeBillingAddress)
+        .then(({ error, token }) => {
+          if (error) {
+            this.setState({
+              isCommittingMutation: false,
+              stripeError: error,
+            })
+          } else {
+            this.createCreditCard({ token: token.id, oneTimeUse: true })
+          }
+        })
+        .catch(e => {
+          this.onMutationError(new ErrorWithMetadata(e))
+        })
+    })
+  }
+
+  @track((props, state, args) => {
```

Comment:
For now we can remove this `track` methods as it'd be instrumenting us against the buy now flow and I'm not sure what the schema is just yet. 😄

Reply:
done